core/services/pipeline/task_params.go

```diff
@@ -232,25 +230,32 @@ type BytesParam []byte
 func (b *BytesParam) UnmarshalPipelineParam(val interface{}) error {
 	switch v := val.(type) {
 	case string:
-		// try hex first
-		if len(v) >= 2 && v[:2] == "0x" {
-			bs, err := hex.DecodeString(v[2:])
+		// first check if this is a valid hex-encoded string
+		if utils.HasHexPrefix(v) {
+			noHexPrefix := utils.RemoveHexPrefix(v)
+			bs, err := hex.DecodeString(noHexPrefix)
 			if err == nil {
-				*b = BytesParam(bs)
+				*b = bs
 				return nil
 			}
-			// The base64 encoding for the binary 0b110100110001 is '0x', so carry on.
 		}
-		// try decoding as base64 first, in case this is a string from the database
-		bs, err := base64.StdEncoding.DecodeString(v)
-		if err != nil {
-			bs = []byte(v)
-		}
-		*b = BytesParam(bs)
-	case []byte:
 		*b = BytesParam(v)
+	case []byte:
+		*b = v
 	case nil:
 		*b = BytesParam(nil)
+
+	case ObjectParam:
+		if v.Type == StringType {
+			*b = BytesParam(v.StringValue)
+			return nil
+		}
+	case *ObjectParam:
+		if v.Type == StringType {
+			*b = BytesParam(v.StringValue)
+			return nil
+		}
```

Comment:
Should an error be returned if not `StringType`?

Reply:
Oh such a great finding! Fixed and also checked all other *Params.